portfolio/src/main/java/com/google/sps/data/Comment.java

```diff
@@ -0,0 +1,11 @@
+package com.google.sps.data;
+
+public final class Comment {
+  private final String value;
+  private final long timeMillis;
+
+  public Comment(Object comment, Object time){
```

Comment:
I would feel a bit more comfortable if the parameters here were String and Long. While making the casts here is convenient, it somewhat falls outside the scope of a constructor.

Here's an idea to keep the casting relatively transparent in calling code:

1. Have a `Comment(String comment, long timeMillis)` constructor.

2. Add a static method to this class which takes in an `Entity` object and attempts to parse out and return a `Comment` (performing any necessary casting).

This allows you to decouple the parsing of an `Entity` from the construction of a `Comment`, and gives you a bit more flexibility in how you handle malformed `Entity` objects. For example, you can add logic to log a malformed `Entity` by checking its parameters are the types you expect with the `instanceof` operator.

In production code, I would consider having this parse method return a `java.util.Optional<Comment>`, to further decouple the error handling and allow calling code to decide what to do. That might be overkill for this case though :)

Reply:
Will do! I definitely need to work on breaking down steps like this especially once we work on testing more.